src/Einsum.jl

```diff
@@ -27,336 +28,404 @@ macro einsum_checkinbounds(ex)
     _einsum(ex, false) # false, false
 end
 
-function _einsum(ex::Expr, inbounds = true, simd = false, threads=false)
+
+function _einsum(expr::Expr, inbounds = true, simd = false, threads = false)
```

Comment:
If changing this, can we make it different to the output expression? `input_expr` or something.